numpy/core/src/npymath/ieee754.c.src

```diff
@@ -574,18 +584,24 @@ npy_longdouble npy_nextafterl(npy_longdouble x, npy_longdouble y)
     defined(__NetBSD__)
 #include <ieeefp.h>
 
-int npy_get_floatstatus(void)
+int npy_get_floatstatus_barrier(char * param))
 {
     int fpstatus = fpgetsticky();
+    /*
+     * By using a volatile, the compiler cannot reorder this call
+     */
+    if (param != NULL) {
```

Comment:
Why do this check, yet pass `x` in from `npy_get_floatstatus`?

Reply:
The correct thing to do is call `npy_get_floatstatus_barrier` directly with a local variable, this currently prevents reordering the call. See for instance `_check_ufunc_fperr` in `numpy/core/src/umath/extobj.c` (where extobj may be NULL), or line 866 in `scalarmath.c.src` which was the original place the reordering was noticed. 

When I fix the documentation from the comment [above](https://github.com/numpy/numpy/pull/11036#pullrequestreview-118975895) I will expand why the `_barrier` form is preferable.

Reply:
I think @eric-wieser was asking why the check was needed at all.